tools/mo/unit_tests/mo/convert/meta_data_test_actual.py

```diff
@@ -0,0 +1,88 @@
+
+
+@generator
+class MetaDataTestTF(unittest.TestCase):
+    test_directory = os.path.dirname(os.path.realpath(__file__))
+
+    @staticmethod
+    def check_meta_data(ov_model, ref_meta):
+        ignore_attrs = ['version', 'optimization']
+        for key, value in ref_meta.items():
+            if key == 'conversion_parameters':
+                for param_name, param_value in value.items():
+                    val = ov_model.get_rt_info([key, param_name])
+                    if param_name in ['extensions', 'caffe_parser_path', 'input_model', 'k', 'output_dir']:
+                        val = Path(val)
+                    assert val == param_value, \
+                        "Runtime info attribute with name {} does not match. Expected: {}, " \
+                        "got {}".format(param_name, param_value, val)
+                continue
+            assert str(ov_model.get_rt_info(key)) == value, \
+                "Runtime info attribute with name {} does not match. Expected: {}, " \
+                "got {}".format(key, value, ov_model.get_rt_info(key))
+
+        for key, value in ov_model.get_rt_info().items():
+            if key in ignore_attrs:
+                continue
+            assert key in ref_meta, "Unexpected runtime info attribute: {}".format(key)
+
+    def test_meta_data_tf(self):
+        def create_tf_model():
+            import tensorflow as tf
+
+            tf.compat.v1.reset_default_graph()
+
+            with tf.compat.v1.Session() as sess:
+                inp1 = tf.compat.v1.placeholder(tf.float32, [1, 2, 3], 'Input')
+                inp2 = tf.compat.v1.placeholder(tf.float32, [1, 2, 3], 'Input')
+                relu = tf.nn.relu(inp1 + inp2, name='Relu')
+
+                output = tf.nn.sigmoid(relu, name='Sigmoid')
+
+                tf.compat.v1.global_variables_initializer()
+                tf_net = sess.graph_def
+            return tf_net
+
+        def ref_meta_data():
+            return {
+                'MO_version': get_version(),
+                'Runtime_version': get_rt_version(),
+                'legacy_frontend': "True",
+            }
+
+        with tempfile.TemporaryDirectory(dir=self.test_directory) as tmpdir:
+
+            model = create_tf_model()
+            out_xml = os.path.join(tmpdir, "model.xml")
+            ref_meta = ref_meta_data()
+
+            ov_model = convert_model(model)
+            self.check_meta_data(ov_model, ref_meta)
+
+            serialize(ov_model, out_xml.encode('utf-8'), out_xml.replace('.xml', '.bin').encode('utf-8'))
+
+            from openvino.runtime import Core
+            core = Core()
+            serialized_model = core.read_model(out_xml)
+            self.check_meta_data(serialized_model, ref_meta)
+
+            restored_graph, meta_data = restore_graph_from_ir(out_xml, out_xml.replace('.xml', '.bin'))
+            save_restored_graph(restored_graph, tmpdir, meta_data, "mo_ir_reader_test_model")
+
```

Comment:
don't we need to check `rt_info` in `restored_graph`?

Reply:
restored_graph is MO networkx based Graph graph obtained from MO IR reader. It has rt_info, but in another format. 
Here I do the following:
1. read model with MO IR reader 
2. serialize with MO IR reader 
3. read IR to ov.Model and check meta data.

So metadata is checked after obtaining ov.Model from it.

Reply:
I understand. We still have POT team work with netwokx based graph. Should we check that this `rt_info` is preserved for them?

Reply:
As meta data is present after serializing/reading of MO IR reader It means that serializing/reading by MO IR reader does not  affect it. As POT uses MO IR reader for serializing/reading such check looks sufficient.